### ISSUE TITLE
run is failing when instance has no tags

### ISSUE THREAD

**Issue by tmichelet** [DataDog/chef-handler-datadog#54]
with the stacktrace 
`
2014-10-10T20:53:55+00:00] ERROR: Report handler Chef::Handler::Datadog raised #<NoMethodError: undefined method 'map!' for nil:NilClass>
`
[full stacktrace here](https://gist.github.com/MrMMorris/2fc8fd9151dac8afa694)
[line causing the issue](https://github.com/DataDog/chef-handler-datadog/blob/master/lib/chef/handler/datadog.rb#L230)

hack was to add `node['tags'] = ['blah']` to avoid this issue
cc @mrmmorris

**Comment by miketheman** [comment]
@MrMMorris @BetaKevin I've been working on a variety of versions and test cases - I have not been able to reproduce this issue. I added a test case that has the conditions of no tags being added in 51f3df76ca73a24086b0173dcd94647d0cd70655 - still wasn't able to suss it out.

Have you been able to reproduce this reliably, or have come up with some other method for showing this error?

**Comment by danieljimenez** [comment]
I'm having this same issue now.

**Comment by miketheman** [comment]
As requested in last comment, and debug details? I cannot reproduce this failure.

**Comment by danieljimenez** [comment]
To work around, all I had to do was add a tag. Here is the stack trace:

```
Running handlers:
[2015-07-21T13:17:46+00:00] ERROR: Report handler Chef::Handler::Datadog raised #<NoMethodError: undefined method `map!' for nil:NilClass>
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-handler-datadog-0.6.0/lib/chef/handler/datadog.rb:230:in `get_node_tags'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-handler-datadog-0.6.0/lib/chef/handler/datadog.rb:215:in `get_combined_tags'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-handler-datadog-0.6.0/lib/chef/handler/datadog.rb:35:in `report'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:226:in `run_report_unsafe'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:214:in `run_report_safely'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:93:in `block in run_report_handlers'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:92:in `each'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:92:in `run_report_handlers'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/handler.rb:103:in `block in <class:Handler>'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/client.rb:416:in `call'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/client.rb:416:in `block in run_completed_successfully'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/client.rb:415:in `each'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/client.rb:415:in `run_completed_successfully'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/client.rb:284:in `run'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:270:in `block in fork_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:258:in `fork'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:258:in `fork_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:224:in `block in run_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:39:in `with_server_connectivity'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:212:in `run_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:402:in `block in interval_run_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:392:in `loop'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:392:in `interval_run_chef_client'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application/client.rb:382:in `run_application'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/lib/chef/application.rb:60:in `run'
[2015-07-21T13:17:46+00:00] ERROR: /opt/chef/embedded/apps/chef/bin/chef-client:26:in `<top (required)>'
[2015-07-21T13:17:46+00:00] ERROR: /usr/bin/chef-client:55:in `load'
[2015-07-21T13:17:46+00:00] ERROR: /usr/bin/chef-client:55:in `<main>'
```

**Comment by miketheman** [comment]
Versions of chef client, chef server?

**Comment by danieljimenez** [comment]
Chef is 12.4.1, Chef server is hosted chef.

**Comment by owen** [comment]
I've seen this as well (being I'm PR #66 :smile:)

I have Chef client 11.12.4, and a hosted Chef environment as well.  I'm going to be spinning up some new nodes shortly and I'll be sure to test node.tags in the chef-shell before I add tags...

**Comment by miketheman** [comment]
Closed via #66.
